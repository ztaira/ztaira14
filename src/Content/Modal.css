.ModalBackground {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
  background-color: rgba(100, 100, 100, 0.5);
  position: fixed;
  top: 0px;
  left: 0px;
}

.ProjectFilterBoundingBox {
  width: 75%;
  max-width: 400px;
  margin: auto;
  position: fixed;
  top: 10%;
}

.ProjectFilters {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  flex-wrap: wrap;
  width: 90%;
  padding: 5%;
  background-color: #fafafa;
}

