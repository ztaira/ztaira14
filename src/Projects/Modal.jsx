import React, { Component } from 'react';
import ProjectFilter from './ProjectFilter.jsx';
import './Modal.css';

class Modal extends Component {
  componentDidMount = () => {
    window.addEventListener('resize', this.ResizeHandlerFunction);
  }

  ResizeHandlerFunction = () => {
    this.FireAfterResize(this.props.modalToggleFunc, 200); 
  }

  FireAfterResize = (func, timeout) => {
    clearTimeout(this.resizeTimer);
    this.resizeTimer = setTimeout(() => { func(); }, timeout);
  }

  ReturnButton = (button) => {
    return (
      <ProjectFilter
        key={button.label}
        button={button}
        muExBtnFunc={this.props.muExBtnFunc}
        modalToggleFunc={this.props.modalToggleFunc}
      />
    );
  }

  componentWillUnmount = () => {
    window.removeEventListener('resize', this.ResizeHandlerFunction);
  }

  render() {
    let shouldDisplay = this.props.active ? "" : "none";
    return (
      <div
        className="ModalBackground"
        style={{
          "width": window.innerWidth,
          "height": window.innerHeight,
          "display": shouldDisplay,
        }}
      >
        <div className="ProjectFilterBoundingBox">
          <div className="ProjectFilters">
            {this.props.buttons.map(this.ReturnButton)}
          </div>
        </div>
      </div>
    );
  }
}

export default Modal;
