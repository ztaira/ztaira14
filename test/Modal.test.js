import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Modal from '../src/Content/Modal.jsx';

it('has not changed since the last snapshot', () => {
  let buttons = [
    {
      'label': 'A-Z',
      'updateFunction': () => {},
      'state': true,
      'updateArgument': () => {},
    },
    {
      'label': 'Z-A',
      'updateFunction': () => {},
      'state': false,
      'updateArgument': () => {},
    },
    {
      'label': 'Newest',
      'updateFunction': () => {},
      'state': false,
      'updateArgument': () => {},
    },
    {
      'label': 'Oldest',
      'updateFunction': () => {},
      'state': false,
      'updateArgument': () => {},
    },
  ];

  const modal = renderer.create(
    <Modal
      buttons={buttons}
      muExBtnFunc={() => {}}
      modalToggleFunc={() => {}}
    />
  );

  let modal_json = modal.toJSON();
  expect(modal_json).toMatchSnapshot();
})
