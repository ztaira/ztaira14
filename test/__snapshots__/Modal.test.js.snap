exports[`test has not changed since the last snapshot 1`] = `
<div
  className="ModalBackground"
  style={
    Object {
      "display": "none",
      "height": 768,
      "width": 1024,
    }
  }>
  <div
    className="ProjectFilterBoundingBox">
    <div
      className="ProjectFilters">
      <button
        className="ProjectFilter"
        id="A-Z"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#666666",
            "borderColor": "#666666",
            "color": "#ffffff",
          }
        }>
        A-Z
      </button>
      <button
        className="ProjectFilter"
        id="Z-A"
        onClick={[Function]}
        style={Object {}}>
        Z-A
      </button>
      <button
        className="ProjectFilter"
        id="Newest"
        onClick={[Function]}
        style={Object {}}>
        Newest
      </button>
      <button
        className="ProjectFilter"
        id="Oldest"
        onClick={[Function]}
        style={Object {}}>
        Oldest
      </button>
    </div>
  </div>
</div>
`;
